internal/cmd/errors_userfacing.go

```diff
@@ -0,0 +1,75 @@
+package cmd
+
+import (
+	"fmt"
+
+	"github.com/infrahq/infra/internal/logging"
+)
+
+// UserFacingError wraps an error to provide better error messages to users.
+// Any non-UserFacingError will be printed as "unhandled errors".
+type UserFacingError struct {
+	// original error
+	Underlying error
+
+	// formatted outputs that users see when running the CLI commands
+	// should be a readable sentence rather than a stacktrace
+	UserFacingMessage string
```

Comment:
Godoc comments always start with the name of the symbol they are documenting, ex:

```
// UserFacingMessage ...
UserFacingMessage string
```

Maybe we can shorten this field name to `Message`, since the type is already called `UserFacingError` ?

```suggestion
	// Message is a human readable error message that is added to Underlying to
	// provider users with more guidance about how to solve the problem.
	Message string
```

Reply:
+1 on the field name though I think even the type can be shortened to `Error` since it exists in the cmd package

Reply:
As in just call the type `Error`

e.g.

```go
if somethingBad {
  return Error{Message: "something bad happened"}
}
```

or 
```go
if err {
  return NewError(err, "something bad happened") # returns Error{Error: err, Message: "..."}
}